crates/sui-core/src/execution_cache/object_locks.rs

```diff
@@ -38,27 +38,19 @@ impl ObjectLocks {
         obj_ref: &ObjectRef,
         epoch_store: &AuthorityPerEpochStore,
     ) -> SuiResult<Option<LockDetails>> {
+        let tables = epoch_store.tables()?;
         match self.locked_transactions.entry(*obj_ref) {
-            DashMapEntry::Vacant(vacant) => {
-                let tables = epoch_store.tables()?;
+            DashMapEntry::Vacant(_) => {
                 let lock = tables.get_locked_transaction(obj_ref)?;
-                if let Some(lock_details) = lock {
-                    vacant.insert(lock_details);
```

Comment:
warming the cache here could actually create a memory leak, removing

Reply:
Please add this comment in code.

Reply:
I'm actually going to go further and remove any reference to in memory state here to make it both simpler and clearer.